apps/web/utils/logic/nextNavigation.ts

```diff
@@ -0,0 +1,143 @@
+// repo
+import { NextNavigationLogic } from "@repo/data/useWalkthroughData";
+// local
+import {
+  WalkthroughAnswerType,
+  WalkthroughStoreAnswers,
+} from "../../stores/AnswerStore";
+
+type NavigationIdType = string | undefined;
+
+export const getNextNavigationId = (
+  nextNavigationLogic: NextNavigationLogic[],
+  answerState: WalkthroughStoreAnswers,
+) => {
+  // setup nextNavigationId variable
+  let nextNavigationId: NavigationIdType;
+
+  // loop through nextNavigationLogic to find the correct next question
+  for (const navigationLogicItem of nextNavigationLogic) {
+    // if fallback item is found, set nextNavigationId to fallback item
+    // paradigms dictate that the fallback item should be the last item in the array
+    if (navigationLogicItem.nextLogicType === "fallback") {
+      nextNavigationId = navigationLogicItem.nextNavigateTo;
+      break;
+    }
+
+    // get answer to check
+    const answerToCheck = answerState[navigationLogicItem.answerToCheck || ""];
+
+    // check if answerToCheck exists
+    if (answerToCheck) {
+      // check if answerValue exists
+      if (navigationLogicItem.answerValue) {
+        /*
+         * NOTE: This logic is only for single answer value strings
+         * there is no support for multiple answers to check in this section
+         */
+
+        // check witch nextLogicType to use
+        switch (navigationLogicItem.nextLogicType) {
```

Comment:
Something we can consider maybe for an future bug proofing optimization is making `nextLogicType` and other "types" like this that are just strings and make them their own enum type. 

This may have been discussed but since we are string matching in a lot of places that definitely leaves room for some typo bugs. Like matching `doesNotContain`  and `doesnotContain`